packages/typescript-estree/src/create-program/createProjectService.ts

```diff
@@ -0,0 +1,70 @@
+import * as ts from 'typescript';
+import * as tsserver from 'typescript/lib/tsserverlibrary';
+
+export function createProjectService() {
+  const compilerOptions = { /* todo */ strict: true };
+  const compilerHost = tsserver.createCompilerHost(compilerOptions, true);
+
+  // TODO: see getWatchProgramsForProjects
+  // We don't watch the disk, we just refer to these when ESLint calls us
+  // there's a whole separate update pass in maybeInvalidateProgram at the bottom of getWatchProgramsForProjects
+  // (this "goes nuclear on TypeScript")
+  const watchFile = (
+    path: string,
+    callback: ts.FileWatcherCallback,
+  ): ts.FileWatcher => {
+    // todo (or just ... stub out?)
+    return { close() {} };
+  };
+
+  const watchDirectory = (
+    path: string,
+    callback: ts.DirectoryWatcherCallback,
+  ) => {
+    // todo (or just ... stub out?)
+    return { close() {} };
+  };
+
+  const system = {
+    ...ts.sys,
```

Comment:
Shocked this works; this is from an entirely different module. I guess it's an interface anyway so it's safeish.